src/Shared-Src/Native/loader.cpp

```diff
@@ -166,7 +165,14 @@ namespace shared {
         std::function<void(const std::string& str)> log_debug_callback,
         std::function<void(const std::string& str)> log_info_callback,
         std::function<void(const std::string& str)> log_warn_callback,
-        const LoaderResourceMonikerIDs& resourceMonikerIDs) {
+        const LoaderResourceMonikerIDs& resourceMonikerIDs,
+        WCHAR const * native_profiler_library_filename) {
+
+        if (native_profiler_library_filename == nullptr)
+        {
+            log_warn_callback("No native profiler library filename was provided. You must pass one to the loader.");

```

Comment:
what does `std::function<void(const std::string& str)> log_debug_callback` expand to? i.e. can `log_debug_callback `be null? if so, we should check before using it

Reply:
Just checked for it - yes, it can be nullptr, so let's check it.

if (perhaps better), we can move the `log_xyz_callback_ = log_xyz_callback;` assignments from further down the ctor to above this logic and then just used the existing `Warn(..)` function

Reply:
✔️